From 96b73c8bda7f1698513846365510d0445476fc8b Mon Sep 17 00:00:00 2001
From: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Date: Fri, 1 Feb 2019 20:13:34 +0530
Subject: [PATCH 0145/1851] ASoC: xlnx: add audio support in 4k mode

commit ffc6544c36ed8a184afc4c33f40c07f22e23a344 from
https://github.com/Xilinx/linux-xlnx.git

Added support for audio enablement in 4k video mode.
1920x1080 and 2048x1080 are the resolutions of sub images for
3840x2160 and 4096x2160 resolutions. Same register programming
applies to those sets.

Signed-off-by: Maruthi Srinivas Bayyavarapu <maruthi.srinivas.bayyavarapu@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 sound/soc/xilinx/xlnx_sdi_audio.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/sound/soc/xilinx/xlnx_sdi_audio.c b/sound/soc/xilinx/xlnx_sdi_audio.c
index ed6908c0e106..a391378bc734 100644
--- a/sound/soc/xilinx/xlnx_sdi_audio.c
+++ b/sound/soc/xilinx/xlnx_sdi_audio.c
@@ -245,15 +245,21 @@ static int xlnx_sdi_tx_hw_params(struct snd_pcm_substream *substream,
 		return -EINVAL;
 	}
 
-	/* map video properties */
+	/*
+	 * map video properties.
+	 * Note: 1920x1080 and 2048x1080 are the resolutions of sub images for
+	 * 3840x2160 and 4096x2160 resolutions respectively.
+	 */
 	switch (ctx->video_mode->hdisplay) {
 	case 1920:
+	case 3840:
 		val = SDI_TRANSPORT_FAMILY_1920;
 		break;
 	case 1280:
 		val |= SDI_TRANSPORT_FAMILY_1280;
 		break;
 	case 2048:
+	case 4096:
 		val |= SDI_TRANSPORT_FAMILY_2048;
 		break;
 	case 720:
-- 
2.31.1

